gravity-forms/gw-disable-submit-button-until-required-fields-are-filled-out.php

```diff
@@ -88,6 +88,12 @@ public function output_script() {
 						$(document).on( 'gppa_updated_batch_fields', function() {
 							self.runCheck();
 						});
+						// Check when field is image choice.
+						gform.addAction( 'gform_input_change', function( elem, formId, fieldId ) {
+							if ( args.inputHtmlIds.includes( `#input_${formId}_${fieldId}` )  &&  $( elem ).parents( '.gfield--type-image_choice' ) ) {
```

Comment:
I'd just put `self.runCheck()` in this action to cover more bases. No need for Image Choice check IMO.

If performance becomes an issue, you could add a super simple debounce around `runCheck` such as:

```js
function gwDisableSubmitDebounce(func, wait, immediate) {
	var timeout;
	return function() {
		var context = this, args = arguments;
		var later = function() {
			timeout = null;
			if (!immediate) func.apply(context, args);
		};
		var callNow = immediate && !timeout;
		clearTimeout(timeout);
		timeout = setTimeout(later, wait);
		if (callNow) func.apply(context, args);
	};
};
```

Reply:
What do we now do with the `onchange` event? Keep or remove it?
```
$( args.inputHtmlIds.join( ', ' ) ).change( function() {
      self.runCheck();
} );
```

Reply:
Keep it for backwards compatibility and on the off chance that `gform_input_change` doesn't fire.

A debounce of like 5-10ms would result in virtually zero noticeable delay, while preventing the logic from running multiple times when an input value changes.